### ISSUE TITLE
Missing example Pact with Karma + Jasmine

### ISSUE THREAD

**Issue by kSzajo** [pact-foundation/docs.pact.io#240]
On https://docs.pact.io/implementation_guides/javascript#examples there is a link pointing to [Pact with Karma + Jasmine](https://github.com/pact-foundation/pact-js/tree/master/examples/karma/jasmine) which does not exist.

**Comment by YOU54F** [comment]
ahh that is actually an old page, and should be removed, you'll note it doesn't have a side bar as it isn't attached to the main sites nav.

Support was dropped for pact-web and karma as per this comment

https://github.com/pact-foundation/pact-js/issues/626#issuecomment-1166264796

might be worth jumping on pact-slack in the pact-js channel to discuss options

**Comment by kSzajo** [comment]
I would like to introduce pact contract testing in angular application. As support for `pact-web` and `karma-pact` discontinued, my only option is to add `Jest` for just sake of using `Pact`?

**Comment by mefellows** [comment]
> I would like to introduce pact contract testing in angular application. As support for pact-web and karma-pact discontinued, my only option is to add Jest for just sake of using Pact?

Well, Jest is one option, but essentially yes. Pact needs a server-side to work and Pact JS is currently not split out architecturally to support running in a non-Node process. Doing so is a non-trivial amount of work and we have no plans to do it at this stage.

**Comment by kSzajo** [comment]
> 

As angular has its own way of testing using karma and jasmine I think it is worth mentioning somewhere within docs that these are not supported and provide some alternative example solution. It might be hard for a new joiner to set up Pact in angular app as community tutorials around the web include discontinued dependencies. What you think?

**Comment by kSzajo** [comment]
I didn't find this page before, I was looking for a straight angular example. Maybe its worth adding a readme file to https://github.com/pact-foundation/pact-js/tree/master/examples/ to indicate that angular projects need to use Jest?

Mentioned [starter schematic](https://github.com/niklas-wortmann/ngx-pact) at https://github.com/pact-foundation/pact-js/blob/master/docs/troubleshooting.md#usage-with-angular  isn't maintained and contains discontinued dependencies

**Comment by mefellows** [comment]
I'm not sure how much that will aid with discoverability either. Perhaps we need a whole section on use cases within the Pact JS documentation so that it stands out in the navbar?

**Comment by kSzajo** [comment]
Agree. I noticed that so far examples are framework agnostic. I finally managed to set up angular using jest but it took me couple hours to understand Pact V3 API and jest config. Example on main page actually isn't compiling which https://github.com/pact-foundation/pact-js#usage because `Promise` needs to be returned
![image](https://user-images.githubusercontent.com/27929367/227199422-1ff4cbe3-5877-4ae9-9a25-4fe826fb0b6a.png)

Additionally Angular is using Observable instead of Promise which also deters new developers from using Pact in angular
